### ISSUE TITLE
Consul server 1.0.0 stalled after some time.

### ISSUE THREAD

**Issue by sterfield** [hashicorp/consul#3721]
### `consul version` for both Client and Server
Client: `1.0.0`
Server: `1.0.0`

### `consul info` for both Client and Server
Server:
```
agent:
	check_monitors = 0
	check_ttls = 0
	checks = 0
	services = 0
build:
	prerelease =
	revision = 51ea240
	version = 1.0.0
consul:
	bootstrap = false
	known_datacenters = 1
	leader = false
	leader_addr = IP:8300
	server = true
raft:
	applied_index = 77374238
	commit_index = 77374238
	fsm_pending = 0
	last_contact = 38.409µs
	last_log_index = 77374239
	last_log_term = 45
	last_snapshot_index = 77366295
	last_snapshot_term = 45
	latest_configuration = [REDACTED]
	latest_configuration_index = 77365448
	num_peers = 4
	protocol_version = 2
	protocol_version_max = 3
	protocol_version_min = 0
	snapshot_version_max = 1
	snapshot_version_min = 0
	state = Follower
	term = 45
runtime:
	arch = amd64
	cpu_count = 2
	goroutines = 194
	max_procs = 2
	os = linux
	version = go1.9.1
serf_lan:
	coordinate_resets = 0
	encrypted = false
	event_queue = 0
	event_time = 38
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 1200
	members = 28
	query_queue = 0
	query_time = 1
serf_wan:
	coordinate_resets = 0
	encrypted = false
	event_queue = 0
	event_time = 1
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 1
	members = 1
	query_queue = 0
	query_time = 1
```

Client:
```
agent:
	check_monitors = 0
	check_ttls = 0
	checks = 0
	services = 0
build:
	prerelease =
	revision = 51ea240
	version = 1.0.0
consul:
	known_servers = 5
	server = false
runtime:
	arch = amd64
	cpu_count = 2
	goroutines = 64
	max_procs = 2
	os = linux
	version = go1.9.1
serf_lan:
	coordinate_resets = 0
	encrypted = false
	event_queue = 0
	event_time = 38
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 3
	member_time = 1200
	members = 31
	query_queue = 0
	query_time = 1
```

### Operating system and Environment details
Ubuntu 16.04.3 LTS on AWS EC2 t2.medium servers.

### Description of the Issue (and unexpected/desired result)
Yesterday (29th of November) around 13:25UTC, we have rotated all our 5 consul servers. Everything went fine, a new leader was elected, and all the 4 other servers correctly joined the new leader. I also rotated all the vault hosts, and verified that a vault master was elected and that vault was available.

Logs of all 5 consul servers are fine, I can see the election, the rotation of the servers, then the rotation of the vault servers, and finally, regular snapshots (roughly every 30 mins on all servers).
At 21:16 UTC the same day, the ELB in front of the vault servers changed status, from 1 healthy node to 0.

The day after, (we have consul-alerts configured, but we did not get any alerts, from consul nor vault), I checked first vault and all nodes were stalled. By experience, I know that if Vault is in bad shape, it's linked to consul, so I went to consul and saw that consul was stalled too.
Every calls to `kv` for example, were stalled.

I stopped all the consul servers then restarted them one by one, and the cluster raised from the dead. Vault was back online too.

The thing that concerns me _a lot_ is that, all consul servers stopped logging a bit after 21:00 29th of November, until the next day. It's the same PID for all servers (they were not restarted or did not changed), but they stopped logging for hours, and I presume, were stalled the whole time until I restart them.

### Reproduction steps
None, it only happens once on 4 different clusters, and they all were running perfectly fine for 3 weeks.

#### Master logs at that time
```
Nov 29 20:20:09 LEADER consul[1620]:     2017/11/29 20:20:09 [INFO] consul.fsm: snapshot created in 26.479µs
Nov 29 20:20:09 LEADER consul[1620]:     2017/11/29 20:20:09 [INFO] raft: Starting snapshot up to 77348936
Nov 29 20:20:09 LEADER consul[1620]:     2017/11/29 20:20:09 [INFO] snapshot: Creating new snapshot at /var/lib/consul/raft/snapshots/8-77348936-1511986809718.tmp
Nov 29 20:20:09 LEADER consul[1620]: consul.fsm: snapshot created in 26.479µs
Nov 29 20:20:09 LEADER consul[1620]: raft: Starting snapshot up to 77348936
Nov 29 20:20:09 LEADER consul[1620]: snapshot: Creating new snapshot at /var/lib/consul/raft/snapshots/8-77348936-1511986809718.tmp
Nov 29 20:20:09 LEADER consul[1620]:     2017/11/29 20:20:09 [INFO] snapshot: reaping snapshot /var/lib/consul/raft/snapshots/8-77332536-1511983530473
Nov 29 20:20:09 LEADER consul[1620]: snapshot: reaping snapshot /var/lib/consul/raft/snapshots/8-77332536-1511983530473
Nov 29 20:20:09 LEADER consul[1620]:     2017/11/29 20:20:09 [INFO] raft: Compacting logs from 77330496 to 77338698
Nov 29 20:20:09 LEADER consul[1620]: raft: Compacting logs from 77330496 to 77338698
Nov 29 20:20:09 LEADER consul[1620]:     2017/11/29 20:20:09 [INFO] raft: Snapshot to 77348936 complete
Nov 29 20:20:09 LEADER consul[1620]: raft: Snapshot to 77348936 complete
Nov 29 20:48:17 LEADER consul[1620]:     2017/11/29 20:48:17 [INFO] consul.fsm: snapshot created in 29.928µs
Nov 29 20:48:17 LEADER consul[1620]:     2017/11/29 20:48:17 [INFO] raft: Starting snapshot up to 77357137
Nov 29 20:48:17 LEADER consul[1620]:     2017/11/29 20:48:17 [INFO] snapshot: Creating new snapshot at /var/lib/consul/raft/snapshots/8-77357137-1511988497305.tmp
Nov 29 20:48:17 LEADER consul[1620]: consul.fsm: snapshot created in 29.928µs
Nov 29 20:48:17 LEADER consul[1620]: raft: Starting snapshot up to 77357137
Nov 29 20:48:17 LEADER consul[1620]: snapshot: Creating new snapshot at /var/lib/consul/raft/snapshots/8-77357137-1511988497305.tmp
Nov 29 20:48:17 LEADER consul[1620]:     2017/11/29 20:48:17 [INFO] snapshot: reaping snapshot /var/lib/consul/raft/snapshots/8-77340735-1511985215796
Nov 29 20:48:17 LEADER consul[1620]: snapshot: reaping snapshot /var/lib/consul/raft/snapshots/8-77340735-1511985215796
Nov 29 20:48:17 LEADER consul[1620]:     2017/11/29 20:48:17 [INFO] raft: Compacting logs from 77338699 to 77346897
Nov 29 20:48:17 LEADER consul[1620]: raft: Compacting logs from 77338699 to 77346897
Nov 29 20:48:17 LEADER consul[1620]:     2017/11/29 20:48:17 [INFO] raft: Snapshot to 77357137 complete
Nov 29 20:48:17 LEADER consul[1620]: raft: Snapshot to 77357137 complete
Nov 30 13:30:22 LEADER consul[1620]:     2017/11/30 13:30:22 [INFO] serf: EventMemberReap: dead-server-1
Nov 30 13:30:22 LEADER consul[1620]: serf: EventMemberReap: dead-server-1
Nov 30 13:35:37 LEADER consul[1620]:     2017/11/30 13:35:37 [INFO] serf: EventMemberReap: dead-server-2
Nov 30 13:35:37 LEADER consul[1620]: serf: EventMemberReap: dead-server-2
```

#### Server follower
```
Nov 29 20:48:18 SERVER-FOLLOWER consul[1471]: raft: Compacting logs from 77338702 to 77346897
Nov 29 20:48:18 SERVER-FOLLOWER consul[1471]:     2017/11/29 20:48:18 [INFO] raft: Snapshot to 77357137 complete
Nov 29 20:48:18 SERVER-FOLLOWER consul[1471]: raft: Snapshot to 77357137 complete
Nov 29 21:18:29 SERVER-FOLLOWER consul[1471]:     2017/11/29 21:18:29 [INFO] consul.fsm: snapshot created in 27.775µs
Nov 29 21:18:29 SERVER-FOLLOWER consul[1471]:     2017/11/29 21:18:29 [INFO] raft: Starting snapshot up to 77365331
Nov 29 21:18:29 SERVER-FOLLOWER consul[1471]:     2017/11/29 21:18:29 [INFO] snapshot: Creating new snapshot at /var/lib/consul/raft/snapshots/8-77365331-1511990309816.tmp
Nov 29 21:18:29 SERVER-FOLLOWER consul[1471]: consul.fsm: snapshot created in 27.775µs
Nov 29 21:18:29 SERVER-FOLLOWER consul[1471]: raft: Starting snapshot up to 77365331
Nov 29 21:18:29 SERVER-FOLLOWER consul[1471]: snapshot: Creating new snapshot at /var/lib/consul/raft/snapshots/8-77365331-1511990309816.tmp
Nov 29 21:18:29 SERVER-FOLLOWER consul[1471]:     2017/11/29 21:18:29 [INFO] snapshot: reaping snapshot /var/lib/consul/raft/snapshots/8-77348941-1511986812037
Nov 29 21:18:29 SERVER-FOLLOWER consul[1471]: snapshot: reaping snapshot /var/lib/consul/raft/snapshots/8-77348941-1511986812037
Nov 29 21:18:29 SERVER-FOLLOWER consul[1471]:     2017/11/29 21:18:29 [INFO] raft: Compacting logs from 77346898 to 77355091
Nov 29 21:18:29 SERVER-FOLLOWER consul[1471]: raft: Compacting logs from 77346898 to 77355091
Nov 29 21:18:29 SERVER-FOLLOWER consul[1471]:     2017/11/29 21:18:29 [INFO] raft: Snapshot to 77365331 complete
Nov 29 21:18:29 SERVER-FOLLOWER consul[1471]: raft: Snapshot to 77365331 complete
Nov 30 10:12:19 SERVER-FOLLOWER consul[1471]: ==> Newer Consul version available: 1.0.1 (currently running: 1.0.0)
Nov 30 13:45:01 SERVER-FOLLOWER consul[1471]:     2017/11/30 13:45:01 [INFO] serf: EventMemberReap: dead-server
Nov 30 13:45:01 SERVER-FOLLOWER consul[1471]: serf: EventMemberReap: dead-server
```

I have all the logs and I'm ready to post it, but I have to anonymize them first.

Do you have any ideas on what happened ?

Thanks a lot for your help,

**Comment by slackpad** [comment]
Hi @sterfield this looks similar to https://github.com/hashicorp/consul/issues/3700, which was fixed in 1.0.1 where all writes to the cluster can get stalled. If you can [enable the debug endpoint](https://www.consul.io/docs/agent/options.html#enable_debug) in your environment we can take a look at goroutine dumps to confirm what's going on.

**Comment by sterfield** [comment]
Hi,

Oh, excellent. I don't have the debug mode activated, and as I mentioned, this only happened once in 3 week of 1.0.0 running on 4 different cluster, so I cannot say if it was the same issue as #3700.

I'll upgrade to 1.0.1 (I have to do another maintenance anyway), and see if I encounter again this issue.

If so, I'll reopen the issue.

Thanks a lot !
